airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -385,19 +384,23 @@ def execute(self, context: 'Context'):
 
             if self.do_xcom_push:
                 result = self.extract_xcom(pod=self.pod)
-            remote_pod = self.pod_manager.await_pod_completion(self.pod)
+            self.pod_manager.await_pod_completion(self.pod)
         finally:
             self.cleanup(
                 pod=self.pod or self.pod_request_obj,
-                remote_pod=remote_pod,
+                context=context,
             )
         ti = context['ti']
         ti.xcom_push(key='pod_name', value=self.pod.metadata.name)
         ti.xcom_push(key='pod_namespace', value=self.pod.metadata.namespace)
         if self.do_xcom_push:
             return result
 
-    def cleanup(self, pod: k8s.V1Pod, remote_pod: k8s.V1Pod):
+    def cleanup(self, pod: k8s.V1Pod, context):
```

Comment:
```suggestion
    def cleanup(self, pod: k8s.V1Pod, context: 'Context'):
```

Just to maintain the type-hint from the above function